cli/lsp/tsc.rs

```diff
@@ -1735,17 +1733,36 @@ pub fn request(
 #[cfg(test)]
 mod tests {
   use super::*;
+  use crate::http_cache::HttpCache;
+  use crate::lsp::analysis;
   use crate::lsp::documents::DocumentCache;
-
-  fn mock_state_snapshot(sources: Vec<(&str, &str, i32)>) -> StateSnapshot {
+  use crate::lsp::sources::Sources;
+  use std::path::Path;
+  use std::path::PathBuf;
+  use tempfile::TempDir;
+
+  fn mock_state_snapshot(
+    fixtures: Vec<(&str, &str, i32)>,
```

Comment:
```suggestion
    fixtures: &[(&str, &str, i32)],
```
Then you can pass the argument by reference (`&arg`) without having to clone() it first.

(Not that it matters here performance-wise but it means I don't have to stop and think about what exactly gets cloned.)